vividus-plugin-saucelabs/src/main/java/org/vividus/sauce/SauceLabsWebDriverFactory.java

```diff
@@ -0,0 +1,109 @@
+import org.vividus.selenium.IBrowserWindowSizeProvider;
+import org.vividus.selenium.IWebDriverFactory;
+import org.vividus.selenium.VividusWebDriver;
+import org.vividus.selenium.VividusWebDriverFactory;
+import org.vividus.selenium.manager.IWebDriverManager;
+import org.vividus.selenium.manager.IWebDriverManagerContext;
+
+public class SauceLabsWebDriverFactory extends VividusWebDriverFactory
+{
+    private static final String SAUCE_OPTION = "sauce:options";
+
+    private ISauceConnectManager sauceConnectManager;
+    private boolean sauceConnectEnabled;
+
+    public SauceLabsWebDriverFactory(IWebDriverFactory webDriverFactory, IBddRunContext bddRunContext,
+            IWebDriverManagerContext webDriverManagerContext, IProxy proxy,
+            IBrowserWindowSizeProvider browserWindowSizeProvider, IWebDriverManager webDriverManager)
+    {
+        super(webDriverFactory, bddRunContext, webDriverManagerContext, proxy, browserWindowSizeProvider,
+                webDriverManager);
+    }
+
+    @Override protected void configureProxy(DesiredCapabilities desiredCapabilities)
+    {
+    }
+
+    @Override
+    protected void configureVividusWebDriver(VividusWebDriver vividusWebDriver)
+    {
+        DesiredCapabilities desiredCapabilities = vividusWebDriver.getDesiredCapabilities();
+        if (sauceConnectEnabled)
+        {
+            addSauceOption(desiredCapabilities, SauceLabsOptionName.TUNNEL_IDENTIFIER,
+                    sauceConnectManager.getTunnelId());
+        }
+        super.configureVividusWebDriver(vividusWebDriver);
+    }
+
+    @Override
+    protected void setDesiredCapabilities(DesiredCapabilities desiredCapabilities, RunningStory runningStory,
+            Scenario scenario, MetaWrapper metaWrapper)
+    {
+        addSauceOption(desiredCapabilities, SauceLabsOptionName.NAME, runningStory.getName());
+    }
+
+    /**
+     * Add only dynamic sauce:options. Others are situated in profile
+     *
+     * @param desiredCapabilities - driver's desired capabilities
+     * @param key                 - sauce-option key
+     * @param value               - sauce-option value
+     */
+    @SuppressWarnings("unchecked")
+    private void addSauceOption(DesiredCapabilities desiredCapabilities,
+            SauceLabsOptionName key, Object value)
+    {
+        // @formatter:off
+        Optional.ofNullable((HashMap<String, Object>) desiredCapabilities.getCapability(SAUCE_OPTION))
+            .ifPresentOrElse(sauceOptions ->
+                sauceOptions.put(key.getOptionName(), value),
+                () -> desiredCapabilities.setCapability(SAUCE_OPTION, adjustNewSauceOptionsCapability(key, value)));
+        // @formatter:on
+    }
+
+    private Map<String, Object> adjustNewSauceOptionsCapability(SauceLabsOptionName key, Object value)
+    {
+        Map<String, Object> sauceOptions = new HashMap<>();
+        sauceOptions.put(key.getOptionName(), value);
+        return sauceOptions;
```

Comment:
`Map.of(...)`

Reply:
I need a modifiable map for SauceOptions